modules/oeFax/controller/AppDispatch.php

```diff
@@ -0,0 +1,248 @@
+<?php
+/**
+ * Fax SMS Module Member
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2018-2019 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General public License 3
+ */
+
+namespace Modules\oeFax\Controller;
+
+use OpenEMR\Common\Crypto\CryptoGen;
+
+require_once(__DIR__ . "/../../../interface/globals.php");
```

Comment:
This seem odd in a class script. Won't the upstream script include this already?

Reply:
depends on the script. This abstract class(glorified interface) determines which class is going to be instantiated RC or Twilio. So `$clientApp = AppDispatch::getApiService();` is static call to dispatcher which then instantiates the required class. Therefore the class is never instantiated in the calling script.

Reply:
I forgot autoloaders too. from when was a module.

Reply:
@bradymiller I cleaned up my classes. really forgot the use and figured the static call to dispatch wasn't in scope with Globals. We really have to do something about Globals! seems like we have to use when we don't need 90% of import in most all scripts.